posthog/schema_migrations/upgrade_manager.py

```diff
@@ -0,0 +1,23 @@
+from collections.abc import Iterator
+from contextlib import contextmanager
+from typing import TYPE_CHECKING
+
+from posthog.hogql_queries.legacy_compatibility.flagged_conversion_manager import conversion_to_query_based
+from posthog.schema_migrations.upgrade import upgrade
+
+if TYPE_CHECKING:
+    from posthog.models.insight import Insight
+
+
+@contextmanager
+def upgrade_query_and_replace_filters(insight: "Insight") -> Iterator[None]:
```

Comment:
should we just name this `upgrade_query`?
Eventually we won't have filters and this just generically upgrades/migrates a query to the latest spec